src/test/java/com/googlecode/download/maven/plugin/internal/WGetTest.java

```diff
@@ -48,29 +80,195 @@ public void setUp() throws Exception {
         setVariableValueToObject(mojo, "retries", 1);
         setVariableValueToObject(mojo, "settings", new Settings());
         setVariableValueToObject(mojo, "buildContext", buildContext);
+
+        initializer.accept(mojo);
+        return mojo;
     }
 
     private void cleanUp() {
         if (cacheDirectory.exists()) {
             stream(cacheDirectory.listFiles()).forEach(File::delete);
             cacheDirectory.delete();
         }
-        if (outputFile.exists()) {
-            outputFile.delete();
-        }
     }
 
     @Test
     public void testCacheDirectoryNotCreated()
             throws IllegalAccessException, MojoExecutionException, MojoFailureException {
-        setVariableValueToObject(mojo, "skipCache", true);
+        WGet mojo = createMojo(m -> setVariableValueToObject(m, "skipCache", true));
 
         try {
             mojo.execute();
-            assertThat("Cache directory should remmain empty if skipTest is true", cacheDirectory.list(),
+            assertThat("Cache directory should remain empty if skipTest is true", cacheDirectory.list(),
                     Matchers.emptyArray());
         } finally {
             cleanUp();
         }
     }
+
+    @Test
+    public void testCacheInANonExistingDirectory() throws MojoExecutionException, MojoFailureException, IllegalAccessException {
+        try {
+            // mojo.execute() will produce an error since wagonManager remains null
```

Comment:
@ajarmoniuk it would be cool to have the explanation in the test javadoc instead